src/services/link/index.js

```diff
@@ -52,9 +52,13 @@ module.exports = function () {
 			const linkData = {};
 			if (data.toHash) {
 				try {
-					await app.service('hash').create(data).then((generatedHash) => {
-						linkData.hash = generatedHash;
-					});
+					const user = (await app.service('users').find({ query: { email: data.toHash } })).data[0];
```

Comment:
find may result in null, so data[0] might fail, again in L122

Reply:
fixed